### ISSUE TITLE
Call back for A unparalled feature in wxMaxima 19.05.7

### ISSUE THREAD

**Issue by idealkindom** [wxMaxima-developers/wxmaxima#1952]
In wxMaxima 19.05.7, which is the default GUI released in bundle of Maxima 5.43.2(which could be downloaded from sourceforge), there is a unparelled feature dispeared in latter release of wxMaxima.

That is, you could fold all the codes and perform a calculation with very little interruction from unneccessary codes in the process, thus making it a very neat and efficient experience.

Here I attached the calculation screenshot in wxMaxima 19.05.7 and rencent  24.08.0_DevelopemtSnapshot for comparison:
<img width="960" alt="19 05 7" src="https://github.com/user-attachments/assets/c4649872-f287-4fa3-9fab-fc54badf39af">

<img width="959" alt="24 08 0" src="https://github.com/user-attachments/assets/20125dbc-140e-4c07-b7a4-9a7312bfed18">
This could be played and compared with any wxm or wxmx files to see and feel the difference.

BTW I found Ctrl+G to stop the calc somehow can't work in 24.08.0.

**Comment by idealkindom** [comment]
One more word,In wxMaxima 19.05.7, you fold the codes, then it fold always like a mermaid. Only the neccessary input remindings halt the process. It keep obeys the rule setting until the end of the calculation, which is absolutely elegant.

**Comment by gunterkoenigsmann** [comment]
Hmmm... ...will try to look into it: I never use folding so if something changes here I don't even know how it is supposed to look.

**Comment by gunterkoenigsmann** [comment]
Is the problem that folding now hides input and output? I seem to remember that there was a bug report that it doesn't so my question is now if there should be yet another option that allows to choose if it should.

**Comment by idealkindom** [comment]
Thanks a lot. Since I used Maxima with wxMaxima for 10 years, I felt the best using experience is with Maxima 5.43.2 bundled with wxMaxima 19.05.7, after which many unpleasant problems annyed from using for daily calculations.

As for the folding, it's easy to imagine the difference if you have many intermediate inputs need.Then you will feel how fantastic is of the way that version 19.05.7 do the work.

在 2024-09-06 02:08:43，"Gunter Königsman" ***@***.***> 写道：


Hmmm... ...will try to look into it: I never use folding so if something changes here I don't even know how it is supposed to look.

**Comment by richardgobeli** [comment]
I use folding to turn off some calculations and not needing to delete them.

**Comment by idealkindom** [comment]
Effective folding could hide unneccessary code during the calculation, and even till the finish the whole working window of the calculation is very concise and neat.

在 2024-09-06 08:03:55，"Richard Gobeli" ***@***.***> 写道：


I use folding to turn off some calculations and not needing to delete them.

**Comment by gunterkoenigsmann** [comment]
Seems like there are 2 groups of users: Ones that want folded cells not to be evaluated and ones that want them to => we need a config option that chooses if to evaluate folded cells.

**Comment by idealkindom** [comment]
> Seems like there are 2 groups of users: Ones that want folded cells not to be evaluated and ones that want them to => we need a config option that chooses if to evaluate folded cells.

what I used in ver 19 is that though folded it evaluated and kept the sheet very neat.

**Comment by gunterkoenigsmann** [comment]
You are right: Our current practice of unfolding cells because their folded tree is being evaluated doesn't make any sense. 

And I would set the default to evaluate the folded cells along with their parent as folding to stop time-consuming parts from being recalculated requires the worksheet to be carefully constructed so the missing parts don't cause problems

**Comment by idealkindom** [comment]
> You are right: Our current practice of unfolding cells because their folded tree is being evaluated doesn't make any sense.
> 
> And I would set the default to evaluate the folded cells along with their parent as folding to stop time-consuming parts from being recalculated requires the worksheet to be carefully constructed so the missing parts don't cause problems

I felt confused why this good feature was not herited in later updates, which stopped me from updating to any new version just after a try ang unintall at once.

**Comment by gunterkoenigsmann** [comment]
It was the only way to avoid a crash I found that happened on trying to evaluate folded cells. 

And I failed to understand that the unfolding of folded cells on evaluation was what this bug is all about. So when wxMaxima was way more stable I never had the idea to fix that bug. Will do that next week and hope you will be surprised how fast and stable wxMaxima has gotten. The only caveat being: This week so many bugs were fixed that I expect that to have resulted in new bugs. Hope we in the meantime have enough tests in place that there won't be too user-visible ones.

**Comment by idealkindom** [comment]
I tested the new release but to find it doesn't work well as to keep the folded cell while evaluation, it has no response. Another problem is the file →new pop up tips without opening a new window
.
